src/mongocxx/test/spec/util.cpp

```diff
@@ -665,134 +835,161 @@ void run_transaction_operations(document::view test,
 
 void run_transactions_tests_in_file(const std::string& test_path) {
     INFO("Test path: " << test_path);
-    auto test_spec = test_util::parse_test_file(test_path);
+
+    const auto test_spec = test_util::parse_test_file(test_path);
     REQUIRE(test_spec);
-    auto test_spec_view = test_spec->view();
-    auto db_name = test_spec_view["database_name"].get_string().value;
-    auto coll_name = test_spec_view["collection_name"].get_string().value;
-    auto tests = test_spec_view["tests"].get_array().value;
+
+    const auto test_spec_view = test_spec->view();
+    const auto db_name = test_spec_view["database_name"].get_string().value;
+    const auto coll_name = test_spec_view["collection_name"].get_string().value;
+    const auto tests = test_spec_view["tests"].get_array().value;
 
     /* we may not have a supported topology */
-    if (should_skip_spec_test(client{uri{}, test_util::add_test_server_api()}, test_spec_view)) {
+    if (should_skip_spec_test({uri{}, test_util::add_test_server_api()}, test_spec_view)) {
         WARN("File skipped - " + test_path);
         return;
     }
 
     for (auto&& test : tests) {
-        bool fail_point_enabled = (bool)test["failPoint"];
-        auto description = test["description"].get_string().value;
-        INFO("Test description: " << description);
-        if (should_skip_spec_test(client{uri{}, test_util::add_test_server_api()},
-                                  test.get_document().value)) {
-            continue;
-        }
-
-        // Steps 1-6.
-        test_setup(test.get_document().value, test_spec_view);
-
-        // Step 7. "Create a new MongoClient client, with Command Monitoring listeners enabled."
-        options::client client_opts;
-        apm_checker apm_checker;
-        client_opts.apm_opts(apm_checker.get_apm_opts(true /* command_started_events_only */));
-        client_opts = test_util::add_test_server_api(client_opts);
-        client client;
-        if (test["useMultipleMongoses"]) {
-            client = {uri{"mongodb://localhost:27017,localhost:27018"}, client_opts};
-        } else {
-            client = {get_uri(test.get_document().value), client_opts};
-        }
-
-        /* individual test may contain a skipReason */
-        if (should_skip_spec_test(client, test.get_document())) {
-            continue;
-        }
+        const auto description = string::to_string(test["description"].get_string().value);
 
-        options::client_session session0_opts;
-        options::client_session session1_opts;
+        SECTION(description) {
```

Comment:
```suggestion
        INFO("Test description: " << description);
        SECTION(description) {
```

Suggestion is intended to include test description on failure output. Example:

```bash
/Users/kevin.albertson/review/mongo-cxx-driver-1007/src/mongocxx/test/spec/monitoring.cpp:128: FAILED:
explicitly with messages:
  checking for path from environment variable: TRANSACTIONS_LEGACY_TESTS_PATH
  test path is: /Users/kevin.albertson/review/mongo-cxx-driver-1007/data/
  transactions/legacy
  Test path: /Users/kevin.albertson/review/mongo-cxx-driver-1007/data/
  transactions/legacy/error-labels.json
  Test description: do not add UnknownTransactionCommitResult label to
  MaxTimeMSExpired inside transactions
  print_all() := "
```

Which can then be used to select the section to run:
```bash
./cmake-build/src/mongocxx/test/test_transactions_specs \
    "Transactions spec automated tests" \
    --section "Legacy" \
    --section "error-labels.json" \
    --section "do not add UnknownTransactionCommitResult label to MaxTimeMSExpired inside transactions"
```

Reply:
The use of the test description as the section name already ensures the description will be included in failure output as the section name, so this should not be necessary (re-emits the same test description twice). For example:

```cpp
TEST_CASE("Example") {
  SECTION("Success Section") {
    SUCCEED("Deliberate success");
  }

  SECTION("Failure Section") {
    FAIL("Deliberate failure");
  }
}
```
yields the following output (with `-s` to include the successful test in output):
```cpp
-------------------------------------------------------------------------------
Example
  Success Section
-------------------------------------------------------------------------------
main.cpp:9
...............................................................................

main.cpp:10: PASSED:
with message:
  Deliberate success

-------------------------------------------------------------------------------
Example
  Failure Section
-------------------------------------------------------------------------------
main.cpp:13
...............................................................................

main.cpp:14: FAILED:
explicitly with message:
  Deliberate failure

===============================================================================
test cases: 1 | 1 failed
assertions: 2 | 1 passed | 1 failed
```

Reply:
Oh nice, thank you. I missed the section name in the output.